pelican/generators.py

```diff
@@ -382,10 +382,10 @@ def generate_feeds(self, writer):
                     arts,
                     self.context,
                     str(self.settings["CATEGORY_FEED_ATOM"]).format(slug=cat.slug),
-                    self.settings.get(
+                    str(self.settings.get(
                         "CATEGORY_FEED_ATOM_URL",
-                        str(self.settings["CATEGORY_FEED_ATOM"]).format(slug=cat.slug),
-                    ),
+                        self.settings["CATEGORY_FEED_ATOM"])
+                    ).format(slug=cat.slug),
```

Comment:
Nice catch. I agree `format` should be done at the end but technically `str(...)` is only needed for `CATEGORY_FEED_ATOM`, because it can be `pathlib.Path`:

```python
                    self.settings.get(
                        "CATEGORY_FEED_ATOM_URL",
                        str(self.settings["CATEGORY_FEED_ATOM"])
                    ).format(slug=cat.slug),
```

Similarly for the others.